### PR TITLE
Factorize meta vars

### LINKED ISSUE #62: Missing date formatting with export option "Duplicate form meta data..."
@aghaynes @PatrickRWright 
**Describe the bug**
Date strings are not formatted as Dates in secuTrial exports where the Export option "Duplicate form meta data into all tables" has been selected.

**To Reproduce**
1. Load a secuTrial export with "Duplicate form meta data into all tables"
2. Check any table with a Date column
3. Columns that contain dates are coded as int

**Expected behavior**
Date strings are transformed into a Date format. These transformed variables are found in a separate column that ends with ".Date".

**Additional context**
It seems that the merge() in dates_secuTrial.secuTrialdata() function on line 42 is not executed correctly because of duplicated metadata columns.

### LINKED ISSUE #66: Factorization/Dates problem for repetition tables in exports with short names
**Describe the bug**
Factor columns are not converted correctly in repetition tables of exports with short table names. They are fine for long names. Also short names in normal (non-repetition) tables are fine.

**To Reproduce**
1. Load an export with export-options$short_names == TRUE containing a repetition table (@PatrickRWright will provide an example soon) using read_secuTrial().
2. have a look at any emnp repetition table with factors. the .factor column will contain only NA

**Expected behavior**
.factor column would contain factor values as coded in cl table

### PR DESCRIPTION
**Pull request by PatrickRWright**
Fixes #62 and fixes #66

### CODE REVIEW

File: R/dates_secuTrial.R [hunk 1]
```diff
@@ -28,21 +29,38 @@ datetimes_secuTrial <- function(x, ...) UseMethod("datetimes_secuTrial", x)
...
 
-  obs <- lapply(x, function(obj){
+  obs <- lapply(table_names, function(obj) {
+
     # find date variables
     it <- object[[object$export_options$meta_names$items]]
     qu <- object[[object$export_options$meta_names$questions]]
+    # if meta data is duplicated then the additional "formtablename"
+    # in the items table creates a problem and is thus removed here
+    if(object$export_options$duplicate_meta) {
```
Comment (aghaynes) [inline R/dates_secuTrial.R]: this we've done elsewhere by only merging it and qu if meta data isn't duplicated

Comment (PatrickRWright) [reply R/dates_secuTrial.R]: Where exactly? It was failing without this exception.

Comment (aghaynes) [reply R/dates_secuTrial.R]: https://github.com/SwissClinicalTrialOrganisation/secuTrialR/blob/df654368e1aee0016539e1339c79f2127fb2f845/R/labels_secuTrial.R#L39

Comment (PatrickRWright) [reply R/dates_secuTrial.R]: The it table contains the itemtype column and qu does not, so the merge is needed here.

**Pull request merged**